docs/deep-dives/pnpm.md

```diff
@@ -0,0 +1,293 @@
+
+This is a double-edged sword when it comes to using **skuba**,
+as this means that our dependencies such as Prettier and ESLint become nested within a `node_modules/skuba/node_modules` subdirectory,
+where most editor and developer tooling integrations will not know to look.
+
+```
+node_modules
+├── skuba -> ./.pnpm/skuba@7.2.0
+└── .pnpm
+    ├── skuba@7.2.0
+    │   └── node_modules
+    │       └── prettier -> ../../prettier@3.0.0
+    └── prettier@3.0.0
+        └── node_modules
+            └── other-dep -> <store>/other-dep
+```
+
+#### .npmrc
+
+pnpm allows us to specify dependencies to hoist via command line or [`.npmrc`] file.
+However, the number of package patterns we need to hoist may fluctuate,
+so specifying hoist patterns via command line would be difficult to maintain.
+
+The **skuba**-maintained `.npmrc` instructs the following dependencies to be hoisted:
+
+```
+# managed by skuba
+public-hoist-pattern[]="@types*"
+public-hoist-pattern[]="*eslint*"
+public-hoist-pattern[]="*prettier*"
+public-hoist-pattern[]="esbuild"
+public-hoist-pattern[]="jest"
+public-hoist-pattern[]="tsconfig-seek"
+# end managed by skuba
+```
+
+From the previous example, this will produce the following `node_modules` layout which means that our integrations will now be able to find `prettier` in `node_modules/prettier` like before.
+
+```
+node_modules
+├── prettier -> ./.pnpm/prettier@3.0.0
+├── skuba -> ./.pnpm/skuba@7.2.0
+└── .pnpm
+    ├── skuba@7.2.0
+    │   └── node_modules
+    │       └── prettier -> ../../prettier@3.0.0
+    └── prettier@3.0.0
+        └── node_modules
+            └── other-dep -> <store>/other-dep
+```
+
+This presents challenges to build pipelines that synthesise an ephemeral `.npmrc` to access private SEEK packages on the npm registry.
+We deal with this in the migration guide below.
+
+## Migrating to pnpm from Yarn or npm
+
+This migration guide assumes that you scaffolded your project with a **skuba** template.
+
+1. Install **skuba** 7.4.0 or greater
+
+2. Install pnpm
+
+   On macOS you can run the following:
+
+   ```bash
+   brew install pnpm
+   ```
+
+   For other operating systems you can check the [install guide].
+
+3. Add a `packageManager` key to your `package.json`
+
+   ```json
+   "packageManager": "pnpm@8.15.1",
+   ```
+
+4. Create [`pnpm-workspace.yaml`](https://pnpm.io/pnpm-workspace_yaml)
+
+   Skip this step if your project does not use Yarn workspaces.
+
+   ```yaml
+   packages:
+     # all packages in direct subdirs of packages/
+     - 'packages/*'
+   ```
+
+5. Run [`pnpm import`]
+
+   This converts your `yarn.lock` or `package-lock.json` into a `pnpm-lock.yaml` file.
+
+6. Delete the `yarn.lock` or `package-lock.json` file
+
+7. Run `skuba format`
+
+   This will synthesise some managed hoist patterns into `.npmrc`.
+
+8. Include additional hoisting settings into `.npmrc`
```

Comment:
Should we add a note around removing the TODO it adds for you?  https://github.com/SEEK-Jobs/indie-billie-eilish/pull/773/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947R25-R28